reconcile/test/test_gabi_authorized_users.py

```diff
@@ -0,0 +1,125 @@
+import os
+import json
+from typing import Dict, List
+from datetime import date, timedelta
+
+from unittest import TestCase
+from unittest.mock import patch
+import reconcile.queries as queries
+from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+import reconcile.openshift_base as ob
+
+from reconcile.test.fixtures import Fixtures
+from reconcile.utils.oc import ApiClient, OCDeprecated
+import reconcile.gabi_authorized_users as gabi_u
+
+apply = Fixtures('gabi_authorized_users').get_anymarkup('apply.yml')
+delete = Fixtures('gabi_authorized_users').get_anymarkup('delete.yml')
+
+
+class RespMock:
+    def __init__(self, data):
+        self.data = json.dumps(data).encode()
+
+
+def apply_request(method, url, query_params=None, headers=None,
+                  post_params=None, body=None, _preload_content=True,
+                  _request_timeout=None):
+    url_list = url.split('/', 1)
+    cluster = apply.get(url_list[0])
+    data = cluster.get(url_list[1], None)
+    return RespMock(data)
+
+
+def delete_request(method, url, query_params=None, headers=None,
+                   post_params=None, body=None, _preload_content=True,
+                   _request_timeout=None):
+    url_list = url.split('/', 1)
+    cluster = delete.get(url_list[0])
+    data = cluster.get(url_list[1], None)
+    return RespMock(data)
+
+
+def mock_get_gabi_instances(expirationDate: str) -> List[Dict]:
+    gabi_instances = apply['gql_response']
+    gabi_instances[0]['expirationDate'] = str(expirationDate)
+    return gabi_instances
+
+
+@patch.dict(os.environ, {"USE_NATIVE_CLIENT": "True"}, clear=True)
+class TestGabiAuthorizedUser(TestCase):
+
+    @patch.object(queries, 'get_app_interface_settings')
```

Comment:
I'd [autospec](https://docs.python.org/3/library/unittest.mock.html#auto-speccing) every patch unless there is a good reason not to.  Without autospec, your tests could pass for cases where you're passing in the wrong number of arguments, calling methods that don't exist, etc.